compiler/luci-interpreter/src/kernels/Reverse.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ * Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "kernels/Reverse.h"
+#include "kernels/Utils.h"
+#include <tensorflow/lite/kernels/internal/reference/reference_ops.h>
+
+namespace luci_interpreter
+{
+
+namespace kernels
+{
+
+Reverse::Reverse(const Tensor *input, const Tensor *axis, Tensor *output)
+    : Kernel({input, axis}, {output})
+{
+}
+
+void Reverse::configure()
+{
+  assert(axis()->shape().num_dims() == 1);
+  assert(input()->shape().num_elements() >= axis()->shape().num_elements());
+  if (input()->element_type() != DataType::S32 && input()->element_type() != DataType::FLOAT32 &&
+      input()->element_type() != DataType::U8 && input()->element_type() != DataType::S16 &&
+      input()->element_type() != DataType::S64)
+  {
+    throw std::runtime_error("Unsupported input type.");
+  }
+  if (axis()->element_type() != DataType::S32)
+  {
+    throw std::runtime_error("Unsupported axis type.");
+  }
+  if (axis()->shape().num_elements() > 1)
+  {
+    throw std::runtime_error("Current does not support more than 1 axis.");
+  }
+  assert(input()->element_type() == output()->element_type());
+
+  output()->resize(input()->shape());
+}
+
+void Reverse::execute() const
+{
+  int axis_value = getTensorData<int32_t>(axis())[0];
+  if (axis_value < 0 || axis_value >= input()->shape().num_dims())
```

Comment:
It should be checked in `configure`.

Reply:
okay I will move to `configure` sequence.